symbolic-symcache/src/writer.rs

```diff
@@ -511,10 +513,20 @@ where
                 self.header.has_line_records = 1;
             }
 
+            let len = (end_address - start_address) as u16;
+            debug_assert_ne!(len, 0, "Function length must be positive");
+            if len == 0 {
+                continue;
+            }
+
+            // This unwrap cannot fail; if len is zero we have already panicked
+            // or continued.
+            let len = NonZeroU16::new(len).unwrap();
```

Comment:
_nit_: This is functionally equivalent, but you can avoid the `unwrap` if you write like this:

```rust
let len = match NonZeroU16::new(len) {
    Some(len) => len,
    None => continue,
};
```